src/hotspot/share/utilities/elfFile.cpp

```diff
@@ -348,4 +302,1411 @@ ElfStringTable* ElfFile::get_string_table(int index) {
+  uint16_t next_word = 0;
+  uint32_t next_dword = 0;
+  uint64_t next_qword = 0;
+
+  switch (attribute_form) {
+    case DW_FORM_addr:
+      // Move position by the size of an address.
+#ifndef _LP64
+      _reader.move_position(4);
+#else
+      _reader.move_position(8);
+#endif
+      break;
+    case DW_FORM_block2:
+      // New position: length + data length (next_word)
+      if (!_reader.read_word(&next_word) || !_reader.move_position(next_word)) {
+        return false;
+      }
+      break;
+    case DW_FORM_block4:
+      // New position: length + data length (next_dword)
+      if (!_reader.read_dword(&next_dword) || !_reader.move_position(next_dword)) {
+        return false;
+      }
+      break;
+    case DW_FORM_data2:
+    case DW_FORM_ref2:
+      if (!_reader.move_position(2)) {
+        return false;
+      }
+      break;
+    case DW_FORM_data4:
+    case DW_FORM_strp: // 4 bytes in 32-bit DWARF
+    case DW_FORM_ref_addr: // second type of reference: 4 bytes in 32-bit DWARF
+    case DW_FORM_ref4:
+      if (!_reader.move_position(4)) {
+        return false;
+      }
+      break;
+    case DW_FORM_data8:
+    case DW_FORM_ref8:
+    case DW_FORM_ref_sig8: // 64-bit type signature
+      if (!_reader.move_position(8)) {
+        return false;
+      }
+      break;
+    case DW_FORM_string:
+      if (!_reader.read_string()) {
+        return false;
+      }
+      break;
+    case DW_FORM_block:
+    case DW_FORM_exprloc:
+      // New position: length + data length (next_qword).
+      if (!_reader.read_uleb128(&next_qword) || !_reader.move_position(next_qword)) {
+        return false;
+      }
+      break;
+    case DW_FORM_block1:
+      // New position: length + data length (next_byte).
+      if (!_reader.read_byte(&next_byte) || !_reader.move_position(next_byte)) {
+        return false;
+      }
+      break;
+    case DW_FORM_data1:
+    case DW_FORM_ref1:
+    case DW_FORM_flag:
+    case DW_FORM_flag_present:
+      if (!_reader.move_position(1)) {
+        return false;
+      }
+      break;
+    case DW_FORM_sdata:
+    case DW_FORM_udata:
+    case DW_FORM_ref_udata:
+      if (!_reader.read_uleb128(&next_qword)) {
+        return false;
+      }
+      break;
+    case DW_FORM_indirect:
+      // Should not be used and therefore is not supported by this parser.
+      return false;
+    case DW_FORM_sec_offset:
+      if (is_DW_AT_stmt_list_attribute) {
+        // DW_AT_stmt_list has the DW_FORM_sec_offset attribute encoding. Store the result in _debug_line_offset.
+        // 4 bytes for 32-bit DWARF.
+        log_develop_trace(dwarf)("    Name: DW_AT_stmt_list, Form: DW_FORM_sec_offset");
+        log_develop_trace(dwarf)("    Reading .debug_line offset from compilation unit at " PTR32_FORMAT, (uint32_t)_reader.get_position());
+        if (!_reader.read_dword(&_debug_line_offset)) {
+          return false;
+        }
+        break;
+      } else {
+        if (!_reader.move_position(4)) {
```

Comment:
Instead of hardcoding the `4` for lineptr/loclistptr/macptr/rangelistptr it would be nice to have a "DwarfOffset` constant of that value, since we only support 32 bit DWARF.